lib/runners/processor/detekt.rb

```diff
@@ -0,0 +1,381 @@
+        else
+          pom_dir = Pathname(Dir.home) / analyzer_name
+          extract_detekt_version_for_maven(pom_dir) || unknown_version
+        end
+    end
+
+    def detekt_config
+      @detekt_config
+    end
+
+    def cli_config
+      detekt_config[:cli]
+    end
+
+    def gradle_config
+      detekt_config[:gradle]
+    end
+
+    def maven_config
+      detekt_config[:maven]
+    end
+
+    def run_cli
+      report_id = "xml"
+      report_path = Tempfile.new("detekt-report")
+      
+      args = [
+        cli_baseline,
+        cli_config_files,
+        cli_config_resource,
+        cli_disable_default_rulesets,
+        cli_excludes,
+        cli_input,
+        cli_report(report_id, report_path.path)
+      ].flatten.compact
+
+      _stdout, stderr, status = capture3(analyzer_bin, *args)
+
+      # detekt has some exit codes.
+      # @see https://github.com/arturbosch/detekt/blob/1.6.0/docs/pages/gettingstarted/cli.md
+      case status.exitstatus
+      when 0, 2
+        construct_result(report_id, report_path)
+      when 3 # invalid configuration
+        Results::Failure.new(guid: guid, message: "Your detekt configuration is invalid", analyzer: analyzer)
+      else
+        trace_writer.error stderr.strip
+        raise stderr
+      end
+    end
+
+    def run_gradle
+      capture3("./gradlew", "--quiet", gradle_config[:task])
+      construct_result( 
+        gradle_config[:report_id],
+        gradle_config[:report_path]
+      )
+    end
+
+    def run_maven
+      mvn_options = %w[--quiet --batch-mode -Dmaven.test.skip=true -Dmaven.main.skip=true]
+      capture3("mvn", maven_config[:goal], *mvn_options)
+      construct_result(
+        maven_config[:report_id],
+        maven_config[:report_path]
+      )
+    end
+
+    def construct_result(report_id, report_path)
+      output_file_path = current_dir / report_path
+      if output_file_path.exist?
+        trace_writer.message "Reading output from #{report_path}..."
+        output = output_file_path.read
+        trace_writer.message output
+      else
+        msg = "#{report_path} does not exist. Unexpected error occurred, processing cannot continue."
+        trace_writer.error msg
+        raise msg
+      end
+
+      switch_constructor(report_id, output)
+    end
+
+    def switch_constructor(report_id, output)
+      case report_id
+      when "xml"
+        construct_checkstyle_result(output)
+      when "txt"
+        construct_plain_result(output)
+      end
+    end
+
+    def construct_checkstyle_result(output)
+      document = REXML::Document.new(output)
+      unless document.root
+        msg = "Invalid XML output!"
+        trace_writer.error msg
+        raise msg
+      end
+
+      issues = []
+
+      document.root.each_element("file") do |file|
+        file.each_element do |error|
+          case error.name
+          when "error"
+            regexp = /detekt\.(.*)/
+            rule, = error[:source].match(regexp).captures
```

Comment:
[`delete_prefix`](https://ruby-doc.org/core-2.7.0/String.html#method-i-delete_prefix) looks safe

```suggestion
            rule = error[:source].delete_prefix("detekt.")
```

Regexp is hard to handle...

```ruby
/detekt\.(.*)/.match("foodetekt.abc").captures[0]
#=> "abc"
```